Description: fixed HMTL,CVS output
Author: Nicolas Hennion <nicolashennion@gmail.com>
Origin: backport, https://github.com/nicolargo/glances/commit/7436794fdc871bf733ee63502b9bf134a7e9595b
Bug: https://github.com/nicolargo/glances/issues/120
Applied-Upstream: https://github.com/nicolargo/glances/commit/7436794fdc871bf733ee63502b9bf134a7e9595b
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/glances/glances.py
+++ b/glances/glances.py
@@ -75,10 +75,10 @@
     print(_('# pip install psutil'))
     print()
     sys.exit(1)
-    
+
 try:
     psutil.disk_io_counters()
-except:
+except Exception:
     print(_('PsUtil > 0.4.0 is needed. Glances cannot start.'))
     print()
     print(_('On Ubuntu 12.04 or higher:'))
@@ -88,7 +88,23 @@
     print(_('# pip install psutil'))
     print()
     sys.exit(1)
-    
+
+try:
+    # HTML output
+    import jinja2
+except ImportError:
+    html_lib_tag = False
+else:
+    html_lib_tag = True
+
+try:
+    # CSV output
+    import csv
+except ImportError:
+    cvs_lib_tag = False
+else:
+    csv_lib_tag = True
+
 
 # Classes
 #========
@@ -2479,25 +2495,9 @@
             server_port = arg
         elif opt in ("-o", "--output"):
             if arg.lower() == "html":
-                try:
-                    # HTML output
-                    import jinja2
-                except ImportError:
-                    print(_("Error: Need Jinja2 library to export into HTML"))
-                    print()
-                    print(_("Try to install the python-jinja2 package"))
-                    sys.exit(2)
-                else:
-                    html_tag = True
+                html_tag = True
             elif arg.lower() == "csv":
-                try:
-                    # CSV output
-                    import csv
-                except ImportError:
-                    print(_("Error: Need CSV library to export to CSV"))
-                    sys.exit(2)
-                else:
-                    csv_tag = True
+                csv_tag = True
             else:
                 print(_("Error: Unknown output %s" % arg))
                 printSyntax()
@@ -2538,6 +2538,11 @@
             sys.exit(2)
     
     if html_tag:
+        if not html_lib_tag:
+            print(_("Error: Need Jinja2 library to export into HTML"))
+            print()
+            print(_("Try to install the python-jinja2 package"))
+            sys.exit(2)
         try:
             output_folder
         except UnboundLocalError:
@@ -2546,6 +2551,9 @@
             sys.exit(2)
 
     if csv_tag:
+        if not csv_lib_tag:
+            print(_("Error: Need CSV library to export into CSV"))
+            sys.exit(2)
         try:
             output_file
         except UnboundLocalError:
